src/App.test.jsx

```diff
@@ -0,0 +1,65 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import App from './App';
+
+test('ensures adding, deleting, editing, and display functionality', () => {
+  render(<App />);
+
+  // Lines 10-23 testing for add item functionality
+
+  // looks for input and add button
+  const addInput = screen.getByRole('textbox');
+  const addButton = screen.getByRole('button', { name: /add/i });
+
+  // user types new item into input
+  const newItem = 'Cheese';
+  userEvent.type(addInput, newItem);
+
+  // user clicks add button
+  userEvent.click(addButton);
+
+  // new item is displayed as a list item
+  const cheeseItem = screen.getByText(/cheese/i);
+  expect(cheeseItem).toBeInTheDocument();
```

Comment:
As Dan mentioned in lecture today, line 22 is an assertion itself, so line 23 isn't needed, but totally acceptable if you like it for readability.